### DISCUSSION TITLE
How can I send Gratuitous ARP using Mongoose with TCP/IP stack enabled?

### DISCUSSION THREAD

**Discussion by covsh** [cesanta/mongoose#2989] [Ask questions, get answers] [answered]
Hi,
Is there any way to send Gratuitous ARP with Mongoose TCP/IP enabled?  
After looking **net_buildin.c** I see that only a direct call to **ether_output** can perform my task. But it has a static scope.

**Comment by scaprile** [comment, marked as answer]
https://github.com/cesanta/mongoose/blob/4a164c39b4ff73d9d46d1df4449e5bc486d080bd/tutorials/tcpip/pcap-driver/main.c#L209-L211

Please see our [documentation](https://mongoose.ws/documentation/),  and follow the guidelines in our [tutorials](https://mongoose.ws/documentation/#tutorials-and-examples).
